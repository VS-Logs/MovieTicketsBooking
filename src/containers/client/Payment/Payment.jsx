import React, { useState } from "react";
import './Payment.scss'

const Form = () => {
  const [paymentData, setpaymentData] = useState({
    bankName: "",
    accNumber: "",
    IFSC: "",
    CVV: ""
  });

  const updatepaymentData = event =>
    setpaymentData({
      ...paymentData,
      [event.target.name]: event.target.value
    });

    const handleFormSubmit=(event)=> {
        alert("Payment Successful!!");
        event.preventDefault();
      }

  const { bankName, accNumber, IFSC, CVV } = paymentData;

  return (<div class="payment">
    <h3>Provide data to make payment </h3><br/>
    <form className="submit" onSubmit={e => handleFormSubmit(e)} id="btn">
     <div>
      <input
        value={bankName}
        onChange={e => updatepaymentData(e)}
        placeholder="Bank Name"
        type="text"
        name="bankName"
        required
      /></div><br/>
      <div><input
        value={accNumber}
        onChange={e => updatepaymentData(e)}
        placeholder="Account Number"
        type="text"
        name="accNumber"
        required
      /></div><br/>
      <div><input
        value={IFSC}
        onChange={e => updatepaymentData(e)}
        placeholder="IFSC address"
        type="IFSC"
        name="IFSC"
        required
      /></div><br/>
      <div><input
        value={CVV}
        onChange={e => updatepaymentData(e)}
        placeholder="CVV"
        type="CVV"
        name="CVV"
        required
      /></div><br/>

<button id="submit" type="submit" value="Submit" >Submit</button>
    </form>
    </div>
  );
};

export default Form;
