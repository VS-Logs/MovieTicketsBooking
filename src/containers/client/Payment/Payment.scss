.body{
    background-color: blue;
}

.payment{
    background-color: darkcyan;
    height: 100vh;
}