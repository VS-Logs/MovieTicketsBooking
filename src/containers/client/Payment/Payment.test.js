import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer'
import Payment from './Payment'
configure({adapter: new Adapter()});

configure({adapter: new Adapter()});
it("renders correctly", () => {
  const wrapper = shallow(
    <Payment />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders a snapshot', () => {
  const tree = renderer.create(<Payment/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should show the text', () => {
    const toggleInstance = shallow(<Payment />);
    const element = toggleInstance.find('div h3');
    expect(element.text()).toMatchSnapshot();
   });