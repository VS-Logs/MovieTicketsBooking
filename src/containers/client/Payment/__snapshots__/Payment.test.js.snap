// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  class="payment"
>
  <h3>
    Provide data to make payment 
  </h3>
  <br />
  <form
    className="submit"
    onSubmit={[Function]}
  >
    <div>
      <input
        name="bankName"
        onChange={[Function]}
        placeholder="Bank Name"
        required={true}
        type="text"
        value=""
      />
    </div>
    <br />
    <div>
      <input
        name="accNumber"
        onChange={[Function]}
        placeholder="Account Number"
        required={true}
        type="text"
        value=""
      />
    </div>
    <br />
    <div>
      <input
        name="IFSC"
        onChange={[Function]}
        placeholder="IFSC address"
        required={true}
        type="IFSC"
        value=""
      />
    </div>
    <br />
    <div>
      <input
        name="CVV"
        onChange={[Function]}
        placeholder="CVV"
        required={true}
        type="CVV"
        value=""
      />
    </div>
    <br />
    <input
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;

exports[`should show the text 1`] = `"Provide data to make payment "`;
